Sources/Data/Microsoft.Psi.Data/Dataset.cs

```diff
@@ -412,6 +466,23 @@ public void AddSessionsFromPsiStores(string path, string partitionName = null)
             {
                 this.AddSessionFromPsiStore(store.Name, store.Path, store.Session, partitionName);
             }
+
+            this.OnChangingOperation();
+        }
+
+        /// <summary>
+        /// Operation call upon any dataset changing operations.
+        /// </summary>
+        public void OnChangingOperation()

```

Comment:
I wonder if we should turn this into an actual event on the Dataset, so other consumers can hook up into it one day. In that case I would also call it OnChanging()

Reply:
If we decide to do that, then perhaps Session should also have an OnChanging() that could fire the Dataset?.OnChanging(). I guess also technically this should be an OnChanged() rather than OnChanging() (the -ing version I think is supposed to fire before the change happens, and the -ed version after)

Reply:
I like the event idea, that would make linking up with PsiStudio easier. I'll try it out and see if there will be any unintended race condition, etc.

Reply:
I came up with two different event designs and I'm not too sure which one is preferable.
1. Each operation invokes the event which the save operation is also subscribed to. The save operation is just another subscribed event.
2. Keep the current structure and the event is only invoked in the `OnChangingOperation`. 

I'm leaning towards (1) since its cleaner.